tools/apigee-envoy-quickstart/aekitctl.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/bash
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+usage() {
+    echo -e "$*\n usage: $(basename "$0")" \
+        "-t <type> -a <action>\n" \
+        "example: $(basename "$0") -t istio-apigee-envoy -a install\n" \
+        "example: $(basename "$0") -t istio-apigee-envoy -a delete\n" \
+        "example: $(basename "$0") -t standalone-apigee-envoy -a install\n" \
+        "example: $(basename "$0") -t standalone-apigee-envoy -a delete\n" \
+        "Parameters:\n" \
+        "-t --type        : Apigee protected Envoy installation type, valida values 'istio-apigee-envoy'. 'standalone-apigee-envoy'\n" \
+        "-a --action      : Install or Delete action, valid values 'install', 'delete'\n" \
+        "-p --platform    : Standalone install for platform opdk or edge, valid values 'opdk', 'edge'\n"
+    exit 1
+}
+
+init() {
+    export CLUSTER_CTX="gke_${PROJECT_ID}_${CLUSTER_LOCATION}_${CLUSTER_NAME}"
+    export ENVOY_AX_SA="x-apigee-envoy-adapter-2-sa"
+    export CLI_HOME=$ENVOY_HOME/apigee-remote-service-cli
+    export REMOTE_SERVICE_HOME=$ENVOY_HOME/apigee-remote-service-envoy
+    export ENVOY_CONFIGS_HOME=$CLI_HOME/envoy-configs-and-samples
+    export AX_SERVICE_ACCOUNT=$ENVOY_HOME/$ENVOY_AX_SA.json
+    export NAMESPACE="apigee"
+    if [ "$PLATFORM" == 'edge' ]; then
+        export MGMT_HOST="https://api.enterprise.apigee.com"
+    fi
+}
+
+createDir() {
+
+    mkdir $CLI_HOME
+    mkdir $REMOTE_SERVICE_HOME
+
+}
+
+PARAMETERS=()
+
+if [[ $# -ne 4 && $# -ne 6 ]]
+then
+   usage;
+   exit 1;
+fi  
+
+while [[ $# -gt 0 ]]
+do
+    param="$1"
+    case $param in
+        -t|--type)
+        export INSTALL_TYPE="$2"
+        shift
+        shift
+        ;;
+        -a|--action)
+        export ACTION="$2"
+        shift
+        shift
+        ;;
+        -p|--platform)
+        export PLATFORM="$2"
+        shift
+        shift
+        ;;
+        *)
+        PARAMETERS+=("$1")
+        shift
+        ;;
+    esac
+done
+ 
+if [[ -z $INSTALL_TYPE ]]; then
+    usage "installation type is a mandatory field"
+fi
+
+if [[ -z $ACTION ]]; then
+    usage "action is a mandatory field"
+fi
+
+if [ "$PLATFORM" != 'opdk' ] && [ "$PLATFORM" != 'edge' ]
+then
+    gke-gcloud-auth-plugin --version 2>&1 >/dev/null
```

Comment:
Where does `gke-gcloud-auth-plugin` come from?

Reply:
This is (will be) mentioned in the README prerequisite section: The link is having the section that explains the install of gke-gcloud-auth-plugin

"If your GKE cluster is version 1.25+ follow this [link](https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke) for enabling required changes for kubectl authentication"